docs/features/non-source-outputs.md

```diff
@@ -0,0 +1,141 @@
+# Artifact Outputs for Incremental Generators
+
+## Summary
+
+This proposal adds a mechanism that allows [incremental generators](incremental-generators.md) to output non-source artifacts, that is, text or binary files that are emitted to disk alongside the compiled assembly.
+
+### Rationale
+
+There exists a desire for incremental source generators be able to write out a set of accompanying files, alongside the generated sources that they add to the compilation. An example of this is JavaScript interop with .NET code. When exposing a .NET method to JavaScript today it is possible to generate the C# interop source code that enables the call to be made from JavaScript, but the user must still hand author the corresponding JavaScript code. Instead, with this feature the incremental generator could emit both the required C# code alongside a JavaScript file that provides the equivalent JavaScript interop code.
+
+### High Level Design Goals
+
+- Allow Incremental Generators to emit source or binary files to disk alongside the emitted .NET assembly
+- The user can control where the generated artifacts are emitted with a suitable default
```

Comment:
including absolute vs relative positioning?

Reply:
I worry about this point. There are two main scenarios for artifact writing that I have in my head: 

1. The `wwwroot` case. Essentially, the generator author knows exactly where in the file system that they want to drop a file.
2. The _use later in build_ case. The generator author wants to put a file on disk that they'll consume later on in the build process. Or make available for other people to consume. 

The problem with customization for (1) is that it breaks the generator entirely. Somehow have to prevent output customizations from impacting this case else we open the door for breaking generator. 

The problem with (2) is that this customization needs to be communicated in our build output. Basically I imagine I'm writing code that is trying to find the output here. Once user customization gets into the mix somehow that has to be communicated out. Trying to imagine the best way for that. Sending out the full list of artifacts even seems hard cause how do I find artifacts from a particular generator?

Reply:
> Sending out the full list of artifacts even seems hard cause how do I find artifacts from a particular generator?

I recall we previously discussed generators being able to pre-define their outputs via MSBuild as a way to inform the rest of the system? For all the scenarios I'm aware of, the set of generated files is finite and will need to land in a well-defined directory so it feels like it's constrained enough to support pre-defining?

Reply:
@captainsafia This is interesting to me, how would you know what the outputs are going to be? Could you expand a little on the scenario you're looking at using this for?